tests/spring/src/test/java/org/jboss/fuse/wsdl2rest/test/calculator/CalculatorDocLitTest.java

```diff
@@ -44,7 +47,108 @@ public void testJavaClient() throws Exception {
+        File cctxFile = outContextPath.toFile();
+        Assert.assertTrue(cctxFile.exists());
+        
+        // validate that we are overriding the JAX-WS service address from the WSDL file
+        Assert.assertTrue(searchForValue("http://other.com/calcservice?", outContextPath));
+    }
+
+    @Test
+    public void testCalculatorNoServiceInWSDL() throws Exception {
+        
+        File wsdlFile = new File("src/test/resources/calculator/calculatorNoAddress.wsdl");
+        Assert.assertTrue(wsdlFile.exists());
+        
+        Path outpath = Paths.get("target/wsdl2rest/calculatorNoService");
+        Wsdl2Rest wsdl2Rest = new Wsdl2Rest(wsdlFile.toURI().toURL(), outpath);
+        
+        List<EndpointInfo> clazzDefs = wsdl2Rest.process();
+        Assert.assertEquals(1, clazzDefs.size());
+        EndpointInfo clazzDef = clazzDefs.get(0);
+        Assert.assertEquals("org.Example", clazzDef.getPackageName());
+        Assert.assertEquals("ICalculator", clazzDef.getClassName());
+        
+        Path contextPath = Paths.get("camel", "wsdl2rest-camel-context.xml");
+        Path outContextPath = outpath.resolve(contextPath);
+        File cctxFile = outContextPath.toFile();
+        Assert.assertTrue(cctxFile.exists());
+        
+        // validate that we are getting the default JAX-WS service address since there is no
+        // service address in the wsdl 
+        Assert.assertTrue(searchForValue("http://localhost:8080/somepath?", outContextPath));
+    }
+
+    @Test
+    public void testCalculatorNoServiceInWSDLAndJAXWSOverride() throws Exception {
+        
+        File wsdlFile = new File("src/test/resources/calculator/calculatorNoAddress.wsdl");
+        Assert.assertTrue(wsdlFile.exists());
+        
+        Path outpath = Paths.get("target/wsdl2rest/calculatorNoServiceWithOverride");
+        Wsdl2Rest wsdl2Rest = new Wsdl2Rest(wsdlFile.toURI().toURL(), outpath);
+
+        // override the JAX-WS service address
+        wsdl2Rest.setJaxwsAddress(new URL("http://other.com/calc"));
+
+        List<EndpointInfo> clazzDefs = wsdl2Rest.process();
+        Assert.assertEquals(1, clazzDefs.size());
+        EndpointInfo clazzDef = clazzDefs.get(0);
+        Assert.assertEquals("org.Example", clazzDef.getPackageName());
+        Assert.assertEquals("ICalculator", clazzDef.getClassName());
+        
+        Path contextPath = Paths.get("camel", "wsdl2rest-camel-context.xml");
+        Path outContextPath = outpath.resolve(contextPath);
+        File cctxFile = outContextPath.toFile();
+        Assert.assertTrue(cctxFile.exists());
+        
+        // validate that we are overriding the default since no service address exists in WSDL
+        Assert.assertTrue(searchForValue("http://other.com/calc?", outContextPath));
+    }
+
+    private boolean searchForValue(String value, Path filePath) throws Exception {
+        Scanner kb = new Scanner(value);
+    	try {
```

Comment:
what about using try-with-resource notation, it will avoid the finally block to close the scanner

Reply:
The PR has been merged, but I see what you're after there. It works the way it is, but I'll try and remember that for next time

Reply:
Created #77